### PR TITLE
Update plugin

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR introduces the update plugin action for it's "active" and "autoUpdateEnabled" states.

/cc @tonyr59h

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PluginSqlUtils.java [hunk 1]
```diff
@@ -37,6 +37,24 @@ private static void removePlugins(@NonNull SiteModel site) {
...
+        }
+
+        PluginModel oldPlugin = getPluginByName(site, plugin.getName());
+        if (oldPlugin == null) {
+            WellSql.insert(plugin).execute();
+            return 1;
+        } else {
+            plugin.setId(oldPlugin.getId());
+            return WellSql.update(PluginModel.class)
+                    .where().equals(PluginModelTable.NAME, plugin.getName())
```
Comment (oguzkocer) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PluginSqlUtils.java]: Would it be better to just use the local id of the `oldPlugin` here instead? I wasn't sure, so I left it this way for now.

File: example/src/test/java/org/wordpress/android/fluxc/plugin/PluginTestUtils.java [hunk 2]
```diff
@@ -22,4 +20,10 @@ static PluginInfoModel generatePluginInfo(String slug) {
         pluginInfo.setSlug(slug);
         return pluginInfo;
     }
+
+    static PluginModel generatePlugin(String name) {
```
Comment (tonyr59h) [inline example/src/test/java/org/wordpress/android/fluxc/plugin/PluginTestUtils.java]: 👍

File: example/src/test/java/org/wordpress/android/fluxc/plugin/PluginStoreUnitTest.java [hunk 3]
```diff
@@ -28,6 +28,7 @@
 import static org.junit.Assert.assertEquals;
 import static org.wordpress.android.fluxc.plugin.PluginTestUtils.generatePluginInfo;
 import static org.wordpress.android.fluxc.plugin.PluginTestUtils.generatePlugins;
+import static org.wordpress.android.fluxc.plugin.PluginTestUtils.generatePlugin;
```
Comment (tonyr59h) [inline example/src/test/java/org/wordpress/android/fluxc/plugin/PluginStoreUnitTest.java]: We may as well just wildcard import from `PluginTestUtils`.

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_PluginTestJetpack.java [hunk 4]
```diff
@@ -69,6 +72,43 @@ public void testFetchPlugins() throws InterruptedException {
         signOutWPCom();
     }
 
+    public void testUpdatePlugin() throws InterruptedException {
+        authenticateWPComAndFetchSites(BuildConfig.TEST_WPCOM_USERNAME_SINGLE_JETPACK_ONLY,
+                BuildConfig.TEST_WPCOM_PASSWORD_SINGLE_JETPACK_ONLY);
+
+        // If a plugin is active and we try to activate it, we'll get an error. If it's not active and we try to
+        // deactivate it, we'll again get an error. So, in order to have a reliable test, we have to fetch the
+        // plugins first, select one and change the activation status and verify that it went through.
```
Comment (tonyr59h) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_PluginTestJetpack.java]: 🤦‍♂️

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PluginSqlUtils.java (hunk 1 above)
Comment (tonyr59h) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PluginSqlUtils.java]: Seems like either way would work. I really like how [PostSqlUtils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PostSqlUtils.java#L26) does it, might be worth doing the same.

**Comment by oguzkocer** [comment]
I can't reply to the `PostSqlUtils` comment inline for some reason. I don't think we need to do it like that, so I am leaving things as is for now. I did make the change to use the wildcard import though. I'll let you merge it as the reviewer. Thanks @tonyr59h!

**Comment by oguzkocer** [comment]
@tonyr59h It turns out wildcards imports are breaking the checkstyle with the following error. I'll just delete that commit.

```
[ant:checkstyle] [ERROR] /home/travis/build/wordpress-mobile/WordPress-FluxC-Android/example/src/test/java/org/wordpress/android/fluxc/plugin/PluginStoreUnitTest.java:29: Using the '.*' form of import should be avoided - org.wordpress.android.fluxc.plugin.PluginTestUtils.*. [AvoidStarImport]
```

**Comment by tonyr59h** [comment]
:shipit:

**Pull request merged**